.github/workflows/ci-pr.yml

```diff
@@ -0,0 +1,48 @@
+# Unique name for this workflow
+name: CI (PR)
+
+# Definition when the workflow should run
+on:
+    pull_request:
+        types: [opened, synchronize, reopened]
+
+# Jobs to be executed
+jobs:
+    formatting-and-linting:
+        runs-on: ubuntu-latest
+        steps:
+            # Checkout the code in the pull request
+            - name: 'Checkout source code'
+              uses: actions/checkout@v1
+
+            # Cache node_modules to speed up the process
+            - name: Restore node_modules cache
+              id: cache-yarn
+              uses: actions/cache@v1
+              with:
+                  path: node_modules
+                  key: yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      yarn-${{ env.cache-name }}-
+                      yarn-
+
+            # Install npm dependencies for Prettier and Jest
+            - name: 'Install npm dependencies'
+              if: steps.cache-yarn.outputs.cache-hit != 'true'
+              run: yarn install
```

Comment:
We should change this to `yarn install --frozen-lockfile`, the counterpart of `npm ci`.